x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -246,6 +249,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (partitionFieldName != null && partitionFieldValue != null && ReservedFieldNames.isValidFieldName(partitionFieldName)) {
             builder.field(partitionFieldName, partitionFieldValue);
         }
+        builder.field(Result.RESULT_TYPE.getPreferredName(), TYPE.getPreferredName());
+        builder.field(MLCATEGORY.getPreferredName(), categoryId);
```

Comment:
As above, actually store it as a string for complete consistency with the results documents.

```suggestion
        builder.field(MLCATEGORY.getPreferredName(), Long.toString(categoryId));
```